ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/admin/ApiConfigManagementService.scala

```diff
@@ -82,23 +83,24 @@ final class ApiConfigManagementService private (
       // Lookup latest configuration to check generation and to extend it with the new time model.
       optConfigAndOffset <- index.lookupConfiguration()
       pollOffset = optConfigAndOffset.map(_._1)
-      currentConfig = optConfigAndOffset
-        .map(_._2)
-        .getOrElse(ledgerConfiguration.initialConfiguration)
+      currentConfig = optConfigAndOffset.map(_._2)
 
       // Verify that we're modifying the current configuration.
-      _ <- if (request.configurationGeneration != currentConfig.generation) {
+      expectedGeneration = currentConfig
```

Comment:
What happens if there are two concurrent calls to `setTimeModel`? Or is this a "preemptive" check and the problem should get caught by the `WriteService`? Is it even a problem?

Reply:
The `WriteService` implementation definitely has the last word on this.